Set up Gemini repo skeleton structure

HamraKhan opened the pull request:
- set up node express with react
- configured vs code to run app locally using one terminal
- added necessary testing libraries/packages

HamraKhan committed:
Set up node Express with react
- configured vs code to run app locally using one terminal
- added necessary testing libraries/packages

navid230 commented on package.json:
```diff
@@ -0,0 +1,45 @@
+  "main": "server.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "concurrently \"nodemon server.js\" \"cd ./client && npm start\"",
```
ensure that you can use it concurrently in a production environment. also check if nodemon is suitable for production running of the server side.

HamraKhan replied:
done :)

navid230 commented on package.json:
```diff
@@ -0,0 +1,45 @@
+  "main": "server.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "concurrently \"nodemon server.js\" \"cd ./client && npm start\"",
```
create a dev script and move the start to dev.

HamraKhan replied:
done :)

navid230 commented on package.json:
```diff
@@ -0,0 +1,45 @@
+    "express": "^4.18.1",
+    "lodash": "^4.17.21",
+    "moment": "^2.29.4",
+    "nodemon": "^2.0.19",
```
move nodemon to devDependencies.

HamraKhan replied:
done :)

navid230 reviewed (changes requested):
please make the appropriate changes, thanks.

navid230 commented:
as discussed, here is a sample implementation of the cluster module. it'll be great if we can use this in the node server
https://blog.appsignal.com/2021/02/03/improving-node-application-performance-with-clustering.html

HamraKhan committed:
GEM-4:
- Added .sample.env file
- Installed cluster and express-rate-limit
- Moved nodemon to devDependencies
- Created a dev script and move the start to dev.
- Added cluster module to spawn multiple processors of the node server.
